### ISSUE TITLE
Godot Crashes On Start-Up if External Editor Cannot be Found

### ISSUE THREAD

**Issue by Capital-EX** [godotengine/godot#10965]
**Operating system and Godot version:**
`Win10 x86_64` with `x86 Godot v2.1.4` from steam

**Issue description:**
Godot will crash if the external editor is unable to be located.  I encountered this bug after moving to Visual Studio Code x64.

**Steps to reproduce:**
1. Enable an external editor
1. Move or delete the editor *.exe
1. Run Godot and open a project
1. Editor will freeze and crash

**Fixed by:**
Manually editing the `editor_data/editor_settings.tres`

**Comment by saltares** [comment]
I tried to reproduce this but was unsuccessful.

**Operating System and Godot version**
Windows 10 64bit with 32bit Godot master 29db531fc8360b1e6d5e23008b208517b6d8c627

It looks like [here](29db531fc8360b1e6d5e23008b208517b6d8c627) we check for `OS::execute()` error code and [here](https://github.com/godotengine/godot/blob/master/platform/windows/os_windows.cpp#L1970) the `OS_Windows::execute()` implementation checks for the error code when trying to open the file and ends up returning `ERR_CANT_FORK`. 

Neither those areas in `os_windows.cpp` and `script_editor_plugin.cpp` seem to have changed recently. So I am reticent to claim this has been fixed. At least, I have been able to reproduce the problem.

**Comment by Capital-EX** [comment]
I found that their is an extra step to causing the error:
1. Enable an external editor
2. Move/delete/rename the editor *.exe
3. Run Godot and open a project
4. ***Grab the window while the Godot is still loading***
5. Editor will freeze and crash

This will present the following error:
```
ERROR: Node::get_tree: Condition ' !data.tree ' is true. returned: 0
   At: .\scene/main/node.h:212
```
It's a bad habit of mine to move windows while programs are still loading.  So, I didn't think it trigger the crash.

**Comment by gregdebonis** [comment]
Wasn't able to reproduce this on Linux. 

All I get is:
**ERROR** OS_Unix::execute - Could not create child process while executing: /home/debonis/editor

- Which is expected. Godot loads just fine.

**Comment by fodinabor** [comment]
Can reproduce it in the Steam version but not in current master.
seems to be fixed by 4b8568006d1f83d518d98ef3c7354b7c25de46cc

**Comment by mhilbrunner** [comment]
This was fixed, see merge above. Close please :)
